lib/engine/part/base.rb

```diff
@@ -7,7 +7,7 @@ module Part
     class Base
       include Helper::Type
 
-      attr_accessor :index, :tile
+      attr_accessor :index, :tile, :lanes
```

Comment:
i don't think lanes should be in base unless every single part has a lane

Reply:
Any part that can be an endpoint of a path should probably have one. I can move it to node, edge and possibly junction.